### PR TITLE
refactor(api-client,core): Use http-status-codes

### PR DESCRIPTION
**Pull request by ffflorian**
## Pull Request Checklist

- [x] My code is covered by tests
- [x] I will [merge the PR as breaking change](https://github.com/wireapp/wire-web-packages/wiki/Releases#create-major-release), if the API contract changes

### CODE REVIEW

**Review by lipis — APPROVED** [review]
💯

File: packages/api-client/src/account/AccountError.ts [hunk 1]
```diff
@@ -17,18 +17,20 @@
  *
  */
 
-import {BackendError, BackendErrorLabel, StatusCode, SyntheticErrorLabel} from '../http';
+import {StatusCodes as HTTP_STATUS} from 'http-status-codes';
```
[the remark this answers is not in the document]
Comment (ffflorian) [reply packages/api-client/src/account/AccountError.ts:20]: Consistency. We usually give `enum`s uppercase names.

File: packages/api-client/src/http/BackendError.ts [hunk 2]
```diff
@@ -17,18 +17,20 @@
...
 export class BackendError extends Error {
-  code: StatusCode;
+  code?: HTTP_STATUS;
   label: BackendErrorLabel | SyntheticErrorLabel;
   message: string;
 
   constructor(
     message: string,
     label: BackendErrorLabel | SyntheticErrorLabel = SyntheticErrorLabel.UNKNOWN,
-    code: StatusCode = StatusCode.UNKNOWN,
```
Comment (ffflorian) [inline packages/api-client/src/http/BackendError.ts:31]: Only caveat: We can't use our custom `UNKNOWN` type anymore.

**Review by bennyn — APPROVED** [review]

File: packages/api-client/src/http/BackendError.ts (hunk 2 above)
Comment (bennyn) [reply packages/api-client/src/http/BackendError.ts:31]: I am okay with that! 👍

**Pull request merged**